src/Common/src/TypeSystem/IL/Stubs/PInvokeILEmitter.cs

```diff
@@ -136,6 +136,24 @@ private MethodIL EmitIL()
             return new  PInvokeILStubMethodIL((ILStubMethodIL)emitter.Link(targetMethod), IsStubRequired());
         }
 
+        public static MethodIL EmitExceptionBody(string message, MethodDesc method)
```

Comment:
There should be a TODO (ideally with an accompanying tracking bug) that makes sure the places that call this:

1. Throw the same exception type the CLR would throw (does interop really throw `System.Exception` when someone messes up their P/invoke declaration?)
2. Get the exception message from the class library. This is needed for glob/loc - the localization of the compiler and the localization of the class library are decoupled. E.g. see `TypeSystemThrowingILEmitter` how we solve this for type system exception messages (e.g. when someone tries to put an explicit structlayout on a generic type and allocate such thing, they'll get a runtime exception with a message that comes from the classlib).
3. We'll need a mechanism by which we can also log a suppressible warning from the compiler if we see this. We'll also want a switch that makes these warnings into errors when needed. (We don't have infra for that right now, but there are a bunch of TODOs - `// TODO: Log as a warning`)